### ISSUE TITLE
Improve provenance information of datasets 

### ISSUE THREAD

**Issue by bernt-matthias** [galaxyproject/galaxy#16344]
**Describe the bug**

It can be hard (or impossible for users) to trace the provenance of datasets if they are copied from other histories. 
Currently there seems to be only the _Inheritance chain_ in the dataset info page:

![Screenshot from 2023-06-30 00-01-35](https://github.com/galaxyproject/galaxy/assets/25689525/8ac8461a-07de-4e2d-86b2-c4d43ae04c6d)

- Since history names are not necessarily unique its insufficient to just list the history name. My _suggestion_ would be to use a link here leading to the corresponding history.
- Same is for dataset names (eg for tools not using the on_strings), i.e. if a history contains multiple datasets with the same name the inheritance chain is of no help. My _suggestion_ would be to include the dataset number (HID?).

**Galaxy Version and/or server at which you observed the bug**
Galaxy Version: 23.0

**Expected behavior**

Dataset provenance is easily traceable .. also if users/tools use suboptimal naming of histories / datasets.

**Comment by mvdbeek** [comment]
Do you know you can walk the job inputs from the info page? This seems more in line with what I would call provenance. What was your goal when you were looking at the inheritance chain ?

**Comment by bernt-matthias** [comment]
> Do you know you can walk the job inputs from the info page? 

Ah, OK .. via the info page of the inputs. Very useful. 

> What was your goal when you were looking at the inheritance chain ?

Ultimately I wanted to find out what I have done :) I guess walking the inputs would work. But it could be cool to also see the context where I/Galaxy created the dataset, i.e. the history where I copied the dataset from. 

The important bit is probably that I copied a dataset to a fresh history but can't remember from where. Walking the inputs helps, but they are not from the current history.  I think the possibility to switch to the history from where the dataset came from would help .. ideally just scrolling to the history item in question.

**Comment by mvdbeek** [comment]
Yep, agreed, I think that is relatively simple on the client. In the job info page we know the history and dataset, so we could be able to tell the history panel to switch and scroll to that input.

**Comment by mvdbeek** [comment]
(the provenance OTOH is hard to do since a job can have many inputs, which in turn can have many inputs ...)

**Comment by simonbray** [comment]
cf https://github.com/galaxyproject/galaxy/issues/11092, seems to be a similar discussion
